apps/web/app/(app)/environments/[environmentId]/product/general/components/EditProductNameForm.tsx

```diff
@@ -0,0 +1,107 @@
+"use client";
+
+import { zodResolver } from "@hookform/resolvers/zod";
+import { useRouter } from "next/navigation";
+import { SubmitHandler, useForm } from "react-hook-form";
+import toast from "react-hot-toast";
+import { z } from "zod";
+
+import { TProduct, ZProduct } from "@formbricks/types/product";
+import { Button } from "@formbricks/ui/Button";
+import { Form, FormControl, FormField, FormItem, FormLabel, FormMessage } from "@formbricks/ui/Form";
+import { Input } from "@formbricks/ui/Input";
+
+import { updateProductAction } from "../actions";
+
+type EditProductNameProps = {
+  product: TProduct;
+  environmentId: string;
+  isProductNameEditDisabled: boolean;
+};
+
+const editProductNameSchema = ZProduct.pick({ name: true });
+
+type TEditProductName = z.infer<typeof editProductNameSchema>;
+
+export const EditProductNameForm: React.FC<EditProductNameProps> = ({
+  product,
+  environmentId,
+  isProductNameEditDisabled,
+}) => {
+  const router = useRouter();
+  const form = useForm<TEditProductName>({
+    defaultValues: {
+      name: product.name,
+    },
+    resolver: zodResolver(editProductNameSchema),
+    mode: "onChange",
+  });
+
+  const nameError = form.formState.errors.name?.message;
+  const isSubmitting = form.formState.isSubmitting;
+
+  const updateProduct: SubmitHandler<TEditProductName> = async (data) => {
+    const name = data.name.trim();
+    try {
+      if (nameError) {
+        toast.error(nameError);
+        return;
+      }
+
+      if (name === product.name) {
```

Comment:
question: Is there a way that react hook form can solve this under the hook. Seems a bit too tedious to do this for each field in the form and I am hoping there is an implicit approach for this.